### ISSUE TITLE
Results of MaxPool2d are different using cpu and cuda when dilation is 2, input is transposed numpy array 

### ISSUE THREAD

**Issue by Francis0103** [pytorch/pytorch#76432]
### 🐛 Describe the bug

```
import numpy as np
import torch
np.random.seed(10)
# shape = (1, 5, 5, 3)
# arr = np.random.randn(*shape).transpose(0, 3, 1, 2) * 10                #（1）

shape = (1, 3, 5, 5)
arr = np.random.randn(*shape) * 10                                          #（2）

kernel_size = 3
padding = 0
dilation = (2, 2)
stride = 1
ceil_mode = False
cpu_x = torch.tensor(arr, dtype=torch.float64, device="cpu")
cuda_x = torch.tensor(arr, dtype=torch.float64, device="cuda")
assert np.allclose(cpu_x.detach().cpu().numpy(), cuda_x.detach().cpu().numpy(), 1e-4, 1e-4)
m = torch.nn.MaxPool2d(
    kernel_size=kernel_size,
    stride=stride,
    padding=padding,
    dilation=dilation,
    ceil_mode=ceil_mode,
)
cpu_y = m(cpu_x)
cuda_y = m(cuda_x)
print("cpu y = \n", cpu_y)
print("cuda y = \n", cuda_y)
assert np.allclose(cpu_y.detach().cpu().numpy(), cuda_y.detach().cpu().numpy(), 1e-4, 1e-4)

```
I got result as below:

> cpu y = 
>  tensor([[[[13.3159]],
> 
>          [[10.2827]],
> 
>          [[12.5647]]]], dtype=torch.float64)
> cuda y = 
>  tensor([[[[14.8454]],
> 
>          [[16.7262]],
> 
>          [[23.9470]]]], device='cuda:0', dtype=torch.float64)

When input comes from (1), the result of 'MaxPool2d' using cpu is different from one using cuda. 
When input comes from (2), the results of 'MaxPool2d' using different devices are the same.
Does this have something to do with tensor stride?

### Versions

'1.12.0a0+git0aa3c39'

cc @ezyang @gchanan @zou3519 @ngimel

**Comment by ptrblck** [comment]
I cannot reproduce the issue in `1.12.0.dev20220425` and get:
```
print("cpu y = \n", cpu_y)
cpu y = 
 tensor([[[[13.3159]],

         [[16.7262]],

         [[11.2879]]]], dtype=torch.float64)

print("cuda y = \n", cuda_y)
cuda y = 
 tensor([[[[13.3159]],

         [[16.7262]],

         [[11.2879]]]], device='cuda:0', dtype=torch.float64)
```
While your commit is also quite new, could you update to the latest commit and in case you are still seeing the differences, fill out the environment information, please?

**Comment by Aidyn-A** [comment]
I was able to reproduce both working and non-working case. 
I will take a deeper look at this. To me it seems like an issue somewhere in `DilatedMaxPool2d.cu`

**Comment by Aidyn-A** [comment]
I thought that there was an issue with CUDA kernel for `max_pool2d`. However, after checking it I found out that the error comes from somewhere else. Now I suspect that there is a mismatch in the way the data is stored on the device. I suspect that tensor constructor just copies raw data of `numpy.ndarray` ignoring that there was a transpose operation.

Basically, if one would do something like this: 
```
cuda_x = torch.tensor(arr.copy(), dtype=torch.float64, device="cuda")
```
Everything works perfectly fine, no matter how arr was created.

@ptrblck do you have any ideas?

**Comment by kshitij12345** [comment]
Fixed in #76597
